src/batch-submitter/tx-batch-submitter.ts

```diff
@@ -35,12 +35,18 @@ import {
 } from '..'
 import { RollupInfo, Range, BatchSubmitter, BLOCK_OFFSET } from '.'
 
+export interface AutoFixBatchOptions {
+  fixDoublePlayedDeposits: boolean
+  fixDelayedTimestampAndBlockNumberHardcoded: boolean
+}
+
 export class TransactionBatchSubmitter extends BatchSubmitter {
   protected chainContract: CanonicalTransactionChainContract
   protected l2ChainId: number
   protected syncing: boolean
   protected lastL1BlockNumber: number
   private disableQueueBatchAppend: boolean
+  private autoFixBatchOptions: AutoFixBatchOptions
 
   constructor(
```

Comment:
We need to pull all this stuff out of the constructor and put it into a config object because this constructor is getting crazy